crates/nu-command/src/strings/detect_columns.rs

```diff
@@ -54,48 +57,114 @@ impl Command for DetectColumns {
         call: &Call,
         input: PipelineData,
     ) -> Result<PipelineData, ShellError> {
-        detect_columns(engine_state, stack, call, input)
+        if !call.has_flag(engine_state, stack, "legacy")? {
+            guess_width(engine_state, stack, call, input)
+        } else {
+            detect_columns_old(engine_state, stack, call, input)
+        }
     }
 
     fn examples(&self) -> Vec<Example> {
         vec![
             Example {
-                description: "Splits string across multiple columns",
-                example: "'a b c' | detect columns --no-headers",
+                description: "detect columns by df output",
+                example: r"
+'Filesystem     1K-blocks      Used Available Use% Mounted on
+none             8150224         4   8150220   1% /mnt/c' | detect columns",
+                result: Some(Value::test_list(vec![Value::test_record(record! {
+                    "Filesystem" => Value::test_string("none"),
+                    "1K-blocks" => Value::test_string("8150224"),
+                    "Used" => Value::test_string("4"),
+                    "Available" => Value::test_string("8150220"),
+                    "Use%" => Value::test_string("1%"),
+                    "Mounted on" => Value::test_string("/mnt/c")
+                })])),
+            },
+            Example {
+                description: "Use --legacy parameter if you find default one does not work",
+                example: "'a b c' | detect columns --legacy --no-headers",
                 result: Some(Value::test_list(vec![Value::test_record(record! {
                         "column0" => Value::test_string("a"),
                         "column1" => Value::test_string("b"),
                         "column2" => Value::test_string("c"),
                 })])),
             },
-            Example {
-                description: "",
-                example:
-                    "$'c1 c2 c3 c4 c5(char nl)a b c d e' | detect columns --combine-columns 0..1",
```

Comment:
I don't really like the idea of removing --combine-columns. Can we add it back but only for --legacy?

Reply:
It seems it may not be working right now anymore. It would be nice if this would work. Seems like it should.
```nushell
^ls -lh | detect columns --skip 1 --no-headers --combine-columns 5..7
```

Reply:
It seems that `--combine-columns` doesn't work with `--no-headers` for a long time..
Not really sure what's going on yet

Reply:
I'm not sure what you mean. This works.
```nushell
❯ ^ls -lh | detect columns -n -s 1 -c 5..7 | rename perms links owner group size date name
╭─#──┬───perms────┬─links─┬──owner──┬──group──┬─size─┬─────date─────┬────────name─────────╮
│ 0  │ -rw-r--r-- │ 1     │ fdncred │ fdncred │ 3.4K │ Mar 31 2023  │ CODE_OF_CONDUCT.md  │
│ 1  │ -rw-r--r-- │ 1     │ fdncred │ fdncred │ 12K  │ Jan 18 09:50 │ CONTRIBUTING.md     │
│ 2  │ -rw-r--r-- │ 1     │ fdncred │ fdncred │ 158K │ Mar 14 08:36 │ Cargo.lock          │
│ 3  │ -rw-r--r-- │ 1     │ fdncred │ fdncred │ 6.3K │ Mar 14 08:36 │ Cargo.toml          │
│ 4  │ -rw-r--r-- │ 1     │ fdncred │ fdncred │ 666  │ Sep 7 2023   │ Cross.toml          │
│ 5  │ -rw-r--r-- │ 1     │ fdncred │ fdncred │ 1.1K │ Apr 3 2023   │ LICENSE             │
│ 6  │ -rw-r--r-- │ 1     │ fdncred │ fdncred │ 12K  │ Mar 14 08:36 │ README.md           │
│ 7  │ drwxr-xr-x │ 3     │ fdncred │ fdncred │ 4.0K │ May 24 2023  │ assets              │
│ 8  │ drwxr-xr-x │ 2     │ fdncred │ fdncred │ 4.0K │ Mar 14 08:36 │ benches             │
│ 9  │ drwxr-xr-x │ 33    │ fdncred │ fdncred │ 4.0K │ Mar 1 13:49  │ crates              │
│ 10 │ drwxr-xr-x │ 2     │ fdncred │ fdncred │ 4.0K │ Jan 18 09:50 │ devdocs             │
```

Reply:
Oh, sorry, I get what you mean.. You mean `guess-width` should also work with merge columns

Reply:
yup, that's what I'm saying, the guess-width functionality still needs to be able to merge columns. I think if you can add that, we can land this.